common-npm-packages/packaging-common/nuget/ProductVersionHelper.ts

```diff
@@ -0,0 +1,12 @@
+import {VersionInfo} from "../pe-parser/VersionResource";
+
+export function getVersionFallback(version: VersionInfo): VersionInfo {
+    const productVersion = version.productVersion;
+    if ((productVersion.a === 0) && (productVersion.b === 0) && (productVersion.c === 0) && (productVersion.d === 0)) {
```

Comment:
This might be made more concise by doing something like:

const { a, b, c, d } = productVersion;
if (a === 0 && b === 0 && c === 0 && d === 0) {
  version.productVersion.a = version.fileVersion.a;
  version.productVersion.b = version.fileVersion.b;
  version.productVersion.c = version.fileVersion.c;
  version.productVersion.d = version.fileVersion.d;
}

Reply:
Made this change!